jax/random.py

```diff
@@ -739,12 +740,19 @@ def _truncated_normal(key, lower, upper, shape, dtype) -> jnp.ndarray:
     _check_shape("truncated_normal", shape, np.shape(lower), np.shape(upper))
 
   sqrt2 = np.array(np.sqrt(2), dtype)
-  a = lax.erf(lax.convert_element_type(lower, dtype) / sqrt2)
-  b = lax.erf(lax.convert_element_type(upper, dtype) / sqrt2)
+  lower = lax.convert_element_type(lower, dtype)
+  upper = lax.convert_element_type(upper, dtype)
+  a = lax.erf(lower / sqrt2)
+  b = lax.erf(upper / sqrt2)
   if not jnp.issubdtype(dtype, np.floating):
     raise TypeError("truncated_normal only accepts floating point dtypes.")
-  u = uniform(key, shape, dtype, minval=jnp.finfo(dtype).tiny)
-  return sqrt2 * lax.erf_inv(a + u * (b - a))
+  u = uniform(key, shape, dtype,
+              minval=lax.nextafter(a, np.array(np.inf, dtype=dtype)), maxval=b)
```

Comment:
Why is this `nextafter` required if we are doing the clamping below?

Reply:
I guess it isn't needed. Done.